fix(admission-controller,agent,node-analyzer,rapid-response,registry-scanner,sysdig): Improve KubeVersion Comparisons

aroberts87 opened the pull request:
## What this PR does / why we need it:
In various places across many charts we do compatibility checks against exact Kubernetes versions. To do so, the charts had been casting the raw value of `.Capabilities.KubeVersion.Minor` to an int like the following:

```
(lt (int .Capabilities.KubeVersion.Minor) 25)
```

The bug here is that certain Kubernetes flavors (EKS, microk8s, etc.) will sometimes put a `+` at the end of the minor version. That `+` was causing the cast to an int to silently fail and return a zero, which was causing failure by attempting to deploy constructs that are not supported on incorrect cluster versions.

This change guards against the `+` suffix as well as abstracts away the version check logic into another helper function that can be more easily adjusted if necessary.


## Checklist

- [x] Title of the PR starts with type and scope, (e.g. feat(agent,node-analyzer,sysdig-deploy):)
- [x] Chart Version bumped for the respective charts
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [x] All test files are added in the tests folder of their respective chart and have a "_test" suffix

aroberts87 committed:
Improve KubeVersion Comparisons

In various places across many charts we do compatibility checks against
exact Kubernetes versions. To do so, the charts had been casting the raw value
of `.Capabilities.KubeVersion.Minor` to an int like the following:

```
(lt (int .Capabilities.KubeVersion.Minor) 25)
```

The bug here is that certain Kubernetes flavors (EKS, microk8s, etc.) will
sometimes put a `+` at the end of the minor version. That `+` was causing the
cast to an int to silently fail and return a zero, which was causing failure
by attempting to deploy constructs that are not supported on incorrect cluster
versions.

This change guards against the `+` suffix as well as abstracts away the version
check logic into another helper function that can be more easily adjusted if
necessary.

mavimo commented on charts/admission-controller/tests/psp_test.yaml:
```diff
@@ -0,0 +1,35 @@
+          kind: RoleBinding
+        template: templates/scanner/rolebinding.yaml
+
+  - it: Ensure PSP is not created on k8s >=1.25
+    capabilities:
+      majorVersion: 1
+      minorVersion: 25
+    asserts:
+      - hasDocuments:
+          count: 0
```
Can we add also a check for EKS?
Something like (to be validated)
```yaml
  - it: Ensure PSP is not created on EKS k8s >=1.25
    capabilities:
      majorVersion: 1
      minorVersion: "25+"
    asserts:
      - hasDocuments:
          count: 0
  - it: Ensure PSP is created on EKS k8s <1.25
    capabilities:
      majorVersion: 1
      minorVersion: "24+"
    asserts:
      - hasDocuments:
          count: 1
```

aroberts87 replied:
Done! 🙂

mavimo commented on charts/sysdig/templates/_helpers.tpl:
```diff
@@ -295,3 +295,16 @@ true
 {{- end -}}
+
+{{/* Returns string 'true' if the cluster's kubeVersion is less than the parameter provided, or nothing otherwise
+     Use like: {{ include "sysdig.kubeVersionLessThan" (dict "root" . "major" "<kube_major_to_compare>" "minor" "<kube_minor_to_compare>") }}
```
```suggestion
     Use like: {{ include "sysdig.kubeVersionLessThan" (dict "root" . "major"<kube_major_to_compare> "minor" <kube_minor_to_compare>) }}
```
major and minor should be number, adding quotes might sound like we need to indicate them as string but then we are not casting them as integer into template..

aroberts87 replied:
All set. 🙂

aroberts87 committed:
Add PSP tests for KubeMinor versions with '+' suffixes

aroberts87 committed:
Update kubeVersionLessThan function headers to remove confusion

* The header had implied that the 'major' and 'minor' values needed to be
  strings, but instead they need to be integers. The updated comment removes
  the double quotes, thus no longer implies the values are strings.

mavimo committed:
chore(admission-controller,agent,node-analyzer,registry-scanner,sysdig): add missing tests to check minor version including '+' sign

mavimo committed:
chore(rapid-response): bump the chart version
